specification.md

```diff
@@ -142,22 +142,232 @@ An ingredient group is a group of related ingredients, e.g. the
 ingredients making up one component of a dish. It consists of:
 
 - *optional* a title
-- 1..n ingredients
+- 0..n ingredients
+- 0..n ingredients groups
 
 An ingredient group us represented as follows:
 
 1. A [heading], whose contents are the group's title
-2. A list of ingredients
-
-[heading]: https://spec.commonmark.org/0.28/#atx-headings
-
-
+2. A list of ingredients (can be skipped)
+3. A group may have child groups: Any groups directly following the
+   group that have a heading level greater than the group's heading
+   are parsed as children.
+
+[heading]: https://spec.commonmark.org/0.28/#atx-heading
+
+## The RecipeMD Parsing Algorithms
+
+This section defines the algorithms to parse a file as a RecipeMD
+recipe. The main algorithm is presented first.
```

Comment:
This section sounds like it is compulsory to follow this exact algorithm. Is this intended? I can imagine that some parts can be achieved with different parsing strategies.

The [CommonMark Specification](https://spec.commonmark.org/0.31.2/#appendix-a-parsing-strategy) uses a wording that sound more open, it just explains how the reference implementations work as a suggestion for implementing the specification.

Reply:
Very good point, thank you! It was not intentional on my part to set an exact algorithm to be followed. I'll update the wording to be more open, similar to the CommonMark spec.